decred/tests/unit/dcr/test_addrlib.py

```diff
@@ -0,0 +1,438 @@
+"""
+Copyright (c) 2019-2020, the Decred developers
+See LICENSE for details
+"""
+
+from base58 import b58decode
+
+from decred import DecredError
+from decred.crypto import crypto
+from decred.dcr import addrlib
+from decred.dcr.nets import mainnet, testnet
+from decred.util.encode import ByteArray
+
+
+def test_addresses():
+    class test:
+        def __init__(
+            self,
+            name="",
+            addr="",
+            saddr="",
+            encoded="",
+            valid=False,
+            scriptAddress=None,
+            f=None,
+            net=None,
+            skipComp=False,
+        ):
+            self.name = name
+            self.addr = addr
+            self.saddr = saddr
+            self.encoded = encoded
+            self.valid = valid
+            self.scriptAddress = scriptAddress
+            self.f = f
+            self.net = net
+            # Pubkey addresses are often printed as pkh addresses, making
+            # comparison of the decoded address impossible.
+            self.skipComp = skipComp
```

Comment:
Since we cannot use dataclasses for 3.6 compatibility, my preference is to use a dict for test data like this. The `dict` constructor affords the same data declaration syntax. There's going to be a little more syntactic noise in the code at the end of the test but that's not too bad. Default values can also be handled in the test code. Not a strong preference in this case, feel free to disregard.